app/controllers/admin/programs_controller.rb

```diff
@@ -13,20 +13,28 @@ def update
       @program.assign_attributes(program_params)
       send_mail_on_schedule_public = @program.notify_on_schedule_public?
 
-      if @program.update_attributes(program_params)
-        ConferenceScheduleUpdateMailJob.perform_later(@conference) if send_mail_on_schedule_public
-        redirect_to admin_conference_program_path(@conference.short_title),
-                    notice: 'The program was successfully updated.'
-      else
-        flash[:error] = "Updating program failed. #{@program.errors.to_a.join('. ')}."
-        render :new
+      respond_to do |format|
+        if @program.update_attributes(program_params)
+          ConferenceScheduleUpdateMailJob.perform_later(@conference) if send_mail_on_schedule_public
+          format.html do
```

Comment:
I would prefer to have two respond_to blocks to be more explicit and to not wrap the whole update statement inside it!